fendermint/vm/interpreter/src/fvm/state/ipc.rs

```diff
@@ -270,10 +258,26 @@ impl<DB: Blockstore> GatewayCaller<DB> {
     }
 }
 
-/// Hash some value in the same way we'd hash it in Solidity.
+/// Total amount of tokens to mint as a result of top-down messages arriving at the subnet.
 ///
-/// Be careful that if we have to hash a single struct, Solidity's `abi.encode`
-/// function will treat it as a tuple.
-pub fn abi_hash<T: Tokenize>(value: T) -> [u8; 32] {
-    keccak256(ethers::abi::encode(&value.into_tokens()))
+/// Fees are ignored, which suggests that they will be paid similarly to gas,
+/// from the balance of the sender already present in the currents subnet.
+pub fn tokens_to_mint(msgs: &[ipc_sdk::cross::CrossMsg]) -> TokenAmount {
+    msgs.iter()
+        .fold(TokenAmount::from_atto(0), |mut total, msg| {
+            total += &msg.msg.value;
```

Comment:
We should also add the fee here. It will generally be zero in the current state, but it should become part of the circulating supply if there are fees (even if we haven't decided yet how to distribute them for top-down messages). This is assuming that you are looking at the `CrossMsg` type, which seems to be the case.

Reply:
Okay, so you are saying that if there are `fees` collected in the parent subnet, then they will be taken there from the sender, locked in the subnet actor, and moved into the subnet, where they will be distributed to the subnet validators or something.

Reply:
Correct. They will de deducted from the balance in the parent, increasing the circulating supply for the subnet. Then in the subnet, we can distributed however we want (although I am aware that we are currently not doing anything with them because we only support `fund`).